src/debug.c

```diff
@@ -326,9 +328,12 @@ void _al_trace_suffix(const char *msg, ...)
             static_trace_buffer);
       }
       #endif
+      #ifdef ALLEGRO_WINDOWS
+         (void) OutputDebugString((LPCTSTR) static_trace_buffer);
+      #endif
       static_trace_buffer[0] = '\0';
    }
-   else if (trace_info.trace_file) {
+   if (trace_info.trace_file) {
```

Comment:
Could you make this be `if (trace_info.trace_file && !_al_user_trace_handler) {`? The idea being that if there is a user trace handler, we don't want to write to the file. The `else` used to take care of it, but now since we need to do the above unconditionally,

Reply:
Yeah, it's sort of weird when you install an handler in between and we still have the file open. Then people will be like: where's the allegro.log copy of it - or why is it empty? :)